SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/EditSneakerEntryActivity.java

```diff
@@ -0,0 +1,178 @@
+			mSneaker.setRarity(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_RARITY_EXTRA));
+			mSneaker.setSellingValue(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_SELLING_EXTRA));
+			mSneaker.setThumbnailId(getIntent().getIntExtra(
+					SneakerDetailsActivity.DETAILS_IMGID_EXTRA, 0));
+			mSneaker.setDescription(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_DESC_EXTRA));
+			mSneaker.setCategory(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_CATEGORY_EXTRA));
+			id = getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_UUID_EXTRA);
+		} else if (isFromListActivity) {
+			mSneaker.setBrand(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_BRAND_EXTRA));
+			mSneaker.setName(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_NAME_EXTRA));
+			mSneaker.setRarity(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_RARITY_EXTRA));
+			mSneaker.setSellingValue(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_SELLING_EXTRA));
+			mSneaker.setThumbnailId(getIntent().getIntExtra(
+					SneakerEntryListActivity.LIST_IMGID_EXTRA, 0));
+			mSneaker.setDescription(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_DESC_EXTRA));
+			mSneaker.setCategory(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_CATEGORY_EXTRA));
+			id = getIntent().getStringExtra(SneakerEntryListActivity.LIST_ID_EXTRA);
+		} else {
+			mCategory.setText(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_CATEGORY_EDIT_EXTRA));
+			mSaveButton.setOnClickListener(new View.OnClickListener() {
+
+				@Override
+				public void onClick(View v) {
+					saveSneaker();
+				}
+			});
+			return;
+		}
+
+		mName.setText(mSneaker.getName());
+		mBrand.setText(mSneaker.getBrand());
+		mSellVal.setText(mSneaker.getSellingValue());
+		@SuppressWarnings("deprecation")
+		Drawable imgDrawable = getResources().getDrawable(mSneaker.getThumbnailId());
+		mThumbnail.setImageDrawable(imgDrawable);
+		mRarity.setText(mSneaker.getRarity());
+		mDesc.setText(mSneaker.getDescription());
+		mCategory.setText(mSneaker.getCategory().getName().toString());
+
+		mSaveButton = (Button) findViewById(R.id.btnSaveData);
+		mSaveButton.setOnClickListener(new View.OnClickListener() {
+
+			@Override
+			public void onClick(View v) {
+				ArrayList<Sneaker> s = SneakerDirectory.get(
+						getApplicationContext()).getAllSneakers();
+				for (Sneaker ss : s) {
+
+					if (id.equals(ss.getId().toString())) {
+						Log.i("Edit Activity", "Sneaker found");
+						ss.setName(mName.getText().toString());
+						ss.setBrand(mBrand.getText().toString());
+						ss.setSellingValue(mSellVal.getText().toString());
+						ss.setRarity(mRarity.getText().toString());
+						ss.setDescription(mDesc.getText().toString());
+						ss.setCategory(mCategory.getText().toString());
+						SneakerDirectory.get(getApplicationContext())
+								.saveSneakers();
```

Comment:
You can refactor your above function saveSneaker() to have a Sneaker parameter and that object will be what it will save. You can then use that to replace lines 152-159